src/sipreg/reg.c

```diff
@@ -299,8 +301,10 @@ static int send_handler(enum sip_transp tp, struct sa *src,
 		sa_set_port(src, reg->srcport);
 
 	reg->laddr = *src;
-	err = mbuf_printf(mb, "Contact: <sip:%s@%J%s>;expires=%u%s%s",
+	err = mbuf_printf(mb, "Contact: <sip:%s@%J%s%s%s>;expires=%u%s%s",
 			  reg->cuser, &reg->laddr, sip_transp_param(reg->tp),
+			  reg->cparams ? ";" : "",
+			  reg->cparams ? reg->cparams : "",
```

Comment:
You really want to add custom URI parameters (which are inside the `< >` brackets) and not SIP address parameters like `;expires=...`?

Reply:
> custom URI parameters (which are inside the < > brackets) 
Positive, see https://www.rfc-editor.org/rfc/rfc8599#page-8

```
     REGISTER sip:alice@example.com SIP/2.0
     Via: SIP/2.0/TCP alicemobile.example.com:5060;branch=z9hG4bKnashds7
     Max-Forwards: 70
     To: Alice <sip:alice@example.com>
     From: Alice <sip:alice@example.com>;tag=456248
     Call-ID: 843817637684230@998sdasdh09
     CSeq: 1826 REGISTER
     Contact: <sip:alice@alicemobile.example.com;
       pn-provider=acme;
       pn-param=acme-param;
       pn-prid=ZTY4ZDJlMzODE1NmUgKi0K>
     Expires: 7200
     Content-Length: 0
```

The test case was also to show explicit intent.

> There is typo in the first commit message/PR title.
Could you please point it out? I just ran the message through a dictionary word by word, am confused :)

Reply:
`sigreg` --> `sipreg`

Reply:
These commits would need to be squashed, I can't rewrite history.. would rewrite the title and tag @sreimers

Reply:
Ok haven't github for a while so changing history isn't only allowed but encouraged :smile: I've force pushed to update the commit message.. also matching both PR's message so CI would work.